### ISSUE TITLE
async issue

### ISSUE THREAD

**Issue by dukecyto** [irmen/Pyro4#110]
When I run examples/async/client.py I get the following error:

> enter async server object uri: PYRO:example.async@localhost:XXXXX
> * normal call: (notice the delay)
> result= 20
> 
> * async call:
> Traceback (most recent call last):
>   File "client.py", line 19, in <module>
>     print("result value available?", asyncresult.ready)  # prints False because the server is still 'busy'
> AttributeError: 'int' object has no attribute 'ready'

It seems that async object is just acting like a regular proxy -- it just blocks, and returns the result directly (in this case int, not FutureResult object).
I used latest Pyro4 (4.41).
I confirmed this problem happens on Windows 7 x64 (Python 2.7.11), OSX (Python 2.7.11), and Ubuntu 15.10 (Python 2.7.10).

It is funny how the code works as expected when I paste it to IPython. I confirmed this on Ubuntu and OSX.

**Comment by dukecyto** [comment]
I think I know what is going on, and found a workaround.

Once remote method is called, the available remote methods are appended to the proxy object. This makes async to directly call those methods when invoked.

The workaround is to create the proxy object every time making the async object. This works for client.py example and for my specific application, but this may not be an option when you need to keep the state of the remote object for subsequent async calls.

In conclusion, I am OK for now, but this is clearly a problem in a more general usage and I hope you will come up with a solution in the future.

**Comment by irmen** [comment]
Yup seems like a bug, I'll investigate
